python/lsst/sims/catUtils/mixins/snObject.py

```diff
@@ -0,0 +1,718 @@
+
+        # mag = np.asarray([[mag]])
+        bandpass = bandpassobject
+
+        if photParams is None:
+            photParams = PhotometricParameters()
+
+        # m5 = np.asarray([[m5]])
+        SNR, gamma = calcSNR_m5(magnitude=mag, bandpass=bandpass, m5=m5,
+                                photParams=photParams)
+        return flux / SNR
+
+    def catsimManyBandFluxes(self, time, bandpassDict,
+                             observedBandPassInd=None):
+        """
+        return the flux in the bandpass in units of the flux
+        the AB magnitude reference spectrum would have in the
+        same band.
+
+        Parameters
+        ----------
+        time: mandatory, float
+            MJD at which band fluxes are evaluated
+        bandpassDict: mandatory, `lsst.sims.photUtils.BandpassDict` instance
+        observedBandPassInd : optional, list of integers, defaults to None
+            integer correspdonding to index of the bandpasses used in the
+            observation in the ordered dict bandpassDict
+        Returns
+        -------
+        `~numpy.ndarray` of length =len(observedBandPassInd)
+
+        Examples
+        --------
+        .. note: If there is an unphysical value of sed in
+        the wavelength range, it produces a flux of  `np.nan`
+        """
+        SEDfromSNcosmo = self.SNObjectSED(time=time,
+                                          bandpass=bandpassDict['u'])
+        wavelen_step = np.diff(SEDfromSNcosmo.wavelen)[0]
+        SEDfromSNcosmo.flambdaTofnu()
+        f = SEDfromSNcosmo.manyFluxCalc(bandpassDict.phiArray,
+                                        wavelen_step=wavelen_step,
+                                        observedBandpassInd=observedBandPassInd)
+        return f / 3631.
+
+    def catsimManyBandMags(self, time, bandpassDict,
+                           observedBandPassInd=None):
+        """
+        return the flux in the bandpass in units of the flux
+        the AB magnitude reference spectrum would have in the
+        same band.
+
+        Parameters
+        ----------
+        time: mandatory, float
+            MJD at which band fluxes are evaluated
+        bandpassDict: mandatory, `lsst.sims.photUtils.BandpassDict` instance
+        observedBandPassInd : optional, list of integers, defaults to None
+            integer correspdonding to index of the bandpasses used in the
+            observation in the ordered dict bandpassDict
+        Returns
+        -------
+        `~numpy.ndarray` of length =len(observedBandPassInd)
+
+        Examples
+        --------
+        .. note: If there is an unphysical value of sed in
+        the wavelength range, it produces a flux of  `np.nan`
+        """
+        f = self.catsimManyBandFluxes(time,
+                                      bandpassDict,
+                                      observedBandPassInd)
+
+        return -2.5 * np.log10(f)
+
+    def catsimBandMags(self, bandpassobject, time):
```

Comment:
should this method be called 'catsimBandMag' since it only calculates the magnitude in one bandpass?